core/services/vrf/listener_v2.go

```diff
@@ -249,6 +263,20 @@ func (lsn *listenerV2) getAndRemoveConfirmedLogsBySub(latestHead uint64) map[uin
 	return toProcess
 }
 
+func ready(req pendingRequest, latestHead uint64) bool {
+	// Request is not eligible for fulfillment yet
+	if req.confirmedAtBlock > latestHead {
+		return false
+	}
+
+	expBackoffFactor := math.Pow(backoffFactor, float64(req.retryCount))
+	delay := backoffInitialDelay * time.Duration(expBackoffFactor)
+	if delay > backoffMaxDelay {
+		delay = backoffMaxDelay
+	}
```

Comment:
Small nit: could pull this into a function `calculateBackoff` which could be tested separately.